diff/lcow/lcow.go

```diff
@@ -131,15 +132,22 @@ func (s windowsLcowDiff) Apply(ctx context.Context, desc ocispec.Descriptor, mou
 		r: io.TeeReader(rdr, digester.Hash()),
 	}
 
-	cmd := exec.Command(
-		"runhcs.exe",
-		"tar2vhd",
-		"--scratchpath", path.Join(layer, "sandbox.vhdx"), // TODO: JTERRY75 when the snapshotter changes this to be scratch.vhdx update it here too.
-		"--destpath", path.Join(layer, "layer.vhd"))
+	layerPath := path.Join(layer, "layer.vhd")
+	outFile, err := os.Create(layerPath)
+	if err != nil {
+		return emptyDesc, err
+	}
+	defer outFile.Close()
+	defer func() {
+		if err != nil {
+			outFile.Close()
+			os.Remove(layerPath)
+		}
+	}()
 
-	cmd.Stdin = rc
-	if bytes, err := cmd.CombinedOutput(); err != nil {
-		return emptyDesc, errors.Wrapf(err, "failed to exec runhcs.exe tar2vhd: %s", string(bytes))
+	err = tar2ext4.Convert(rc, outFile, tar2ext4.ConvertWhiteout, tar2ext4.AppendVhdFooter)
+	if err != nil {
+		return emptyDesc, errors.Wrapf(err, "failed to convert tar to ext4 vhd")
```

Comment:
Should we have the size option here too?

Reply:
Done